tests/main.test.ts

```diff
@@ -195,24 +195,85 @@ describe('zip-it-and-ship-it', () => {
     await zipNode('side-module', { opts: options })
   })
 
-  testMany('Ignore some excluded node modules', [...allBundleConfigs], async (options) => {
-    const fixtureName = 'node-module-excluded'
-    const opts = merge(options, {
-      basePath: join(FIXTURES_DIR, fixtureName),
-    })
-    const { files } = await zipNode(fixtureName, { opts })
+  describe('aws-sdk special case', () => {
+    testMany(
+      'On Node v18, includes v2 and excludes v3',
+      ['bundler_default', 'bundler_esbuild', 'bundler_esbuild_zisi', 'bundler_default_nft', 'bundler_nft'],
+      async (options) => {
+        const fixtureName = 'node-module-excluded'
+        const opts = merge(options, {
+          basePath: join(FIXTURES_DIR, fixtureName),
+          config: {
+            '*': {
+              ...options.config['*'],
+              nodeVersion: '18',
+            },
+          },
+        })
+        const { files } = await zipNode(fixtureName, { opts })
+
+        const [{ unzipPath, bundler, entryFilename }] = files
+
+        if (bundler === 'esbuild') {
+          // esbuild bundles everything into one file, so we need to assert on the file contents
+          const bundle = await readFile(`${unzipPath}/${entryFilename}`, 'utf-8')
+          expect(bundle).toContain('/aws-sdk')
+          expect(bundle).not.toContain('/@aws-sdk')
+        } else {
+          await expect(`${files[0].unzipPath}/node_modules/aws-sdk`).toPathExist()
+          await expect(`${files[0].unzipPath}/node_modules/@aws-sdk/client-s3`).not.toPathExist()
+        }
 
-    await expect(`${files[0].unzipPath}/node_modules/aws-sdk`).not.toPathExist()
+        try {
+          const func = await importFunctionFile(`${files[0].unzipPath}/function.js`)
 
-    try {
-      const func = await importFunctionFile(`${files[0].unzipPath}/function.js`)
+          func()
+
+          expect.fail('Running the function should fail due to the missing module')
+        } catch (error) {
+          expect(error.code).toBe('MODULE_NOT_FOUND')
+        }
+      },
+    )
 
-      func()
+    testMany(
+      'On Node v16, excludes v2 and includes v3',
+      ['bundler_default', 'bundler_esbuild', 'bundler_esbuild_zisi', 'bundler_default_nft', 'bundler_nft'],
+      async (options) => {
+        const fixtureName = 'node-module-excluded'
+        const opts = merge(options, {
+          basePath: join(FIXTURES_DIR, fixtureName),
+          config: {
+            '*': {
+              ...options.config['*'],
+              nodeVersion: '16',
+            },
+          },
+        })
+        const { files } = await zipNode(fixtureName, { opts })
+
+        const [{ unzipPath, bundler, entryFilename }] = files
+        if (bundler === 'esbuild') {
```

Comment:
The previous test case silently ignored this case. It still passed, since we only asserted on a file being *nonexistant*. But now we're also asserting on a different file existing.